### PR TITLE
Added Watch later Functionality

### PR DESCRIPTION
**Pull request by Ankur9669**
Files to check
1. Watchlater.js
2. watch-later-context.js
3. watch-later-reducer.js
4. Video.js

### CODE REVIEW

**Comment by Priya-Patel31** [comment]
image


Watch later is missing from sidebar . So could not review as there is no link  present to see watch later page .

**Review by nikhiljugale007 — APPROVED** [review]
Did not find any issues in the code. Good naming convention and clean code. 
Good to merge.

File: FitVid/src/Pages/LikedVideos/LikedVideos.js [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
 import EmptyList from "../../Components/EmptyList/EmptyList";
 
 const LikedVideos = () => {
   const { likedVideos } = useLiked();
+  const { user } = useUser();
+  const { showToast } = useToast();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!user.isUserLoggedIn) {
```
Comment (Priya-Patel31) [inline FitVid/src/Pages/LikedVideos/LikedVideos.js]: Why you have done like this ?? You can achieve this without using inside useEffect.

File: FitVid/src/Util/remove-from-history.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    });
+    return {
+      data: response.data,
+      success: true,
+      message: "History Cleared",
+    };
+  } catch (e) {
+    console.log(e);
+    return {
+      data: "",
```
Comment (Priya-Patel31) [inline FitVid/src/Util/remove-from-history.js]: You can have data as null rather than "" .

File: FitVid/src/Util/remove-from-watch-later.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    });
+    return {
+      data: response.data,
+      success: true,
+      message: "Removed From WatchLater Videos",
+    };
+  } catch (e) {
+    console.log(e);
+    return {
+      data: "",
```
Comment (Priya-Patel31) [inline FitVid/src/Util/remove-from-watch-later.js:19]: same as above

File: FitVid/src/index.js [hunk 4]
```diff
@@ -7,21 +7,27 @@ import { makeServer } from "./server";
...
-    <LikedProvider>
-      <UserProvider>
-        <ToastProvider>
-          <Router>
-            <App />
-          </Router>
-        </ToastProvider>
-      </UserProvider>
-    </LikedProvider>
+    <WatchLaterProvider>
```
Comment (Priya-Patel31) [inline FitVid/src/index.js:19]: You can have one file for all the contexts and import it over here .

**Review by Priya-Patel31 — APPROVED** [review]
Everything seems good to me , good naming conventions have been followed . Just have few minor changes you can go through it .

File: FitVid/src/Pages/LikedVideos/LikedVideos.js (hunk 1 above)
Comment (Ankur9669) [reply FitVid/src/Pages/LikedVideos/LikedVideos.js]: I will confirm this

File: FitVid/src/index.js (hunk 4 above)
Comment (Ankur9669) [reply FitVid/src/index.js:19]: Will do it

**Pull request merged**